src/saveToCase.js

```diff
@@ -0,0 +1,475 @@
+define([
+    'vellum/form',
+    'jquery',
+    'underscore',
+    'vellum/mugs',
+    'vellum/parser',
+    'vellum/tree',
+    'vellum/util',
+    'vellum/widgets',
+    'tpl!vellum/templates/widget_update_case',
+    'vellum/core'
+], function (
+    form_,
+    $,
+    _,
+    mugs,
+    parser,
+    Tree,
+    util,
+    widgets,
+    widget_update_case
+){
+    function createsCase(mug) {
+        return mug ? mug.p.use_create : false;
+    }
+
+    function closesCase(mug) {
+        return mug ? mug.p.use_close : false;
+    }
+
+    function updatesCase(mug) {
+        return mug ? mug.p.use_update : false;
+    }
+
+    function addSetValue(mug) {
+        var path = mug.absolutePath;
+
+        if (createsCase(mug)) {
+            mug.form.addSetValue('xforms-ready', path + "/case/@case_id", mug.p.case_id);
+        }
+    }
+
+    var propertyWidget = function (mug, options) {
+            var widget = widgets.normal(mug, options),
+                id = options.id;
+
+            widget.kvInput = $('<div class="control-row" />').attr('name', id);
+
+            widget.getControl = function () {
+                return widget.kvInput;
+            };
+
+            widget.refreshControl = function () {
+                widget.setValue(widget.getValue());
+            };
+
+            return widget;
+        },
+        saveCasePropWidget = function (mug, options) {
+            var widget = propertyWidget(mug, options);
+
+            widget.setValue = function (value) {
+                value = _.isUndefined(value) ? {} : value;
+                widget.kvInput.html(widget_update_case({
+                    props: value
+                }));
```

Comment:
Yeah, `baseKeyValue` seems pretty old/outdated, judging by [this comment](https://github.com/dimagi/Vellum/blob/master/src/widgets.js#L21-L23) since it falls into the category of widgets that inherit from `base`. I did a bunch of work to standardize and cleanup the `getValue`/`setValue` implementations on a bunch of the widgets, including `xPath`, but I notably did not touch `baseKeyValue` because it was doing things a really different/crufty way.